flycheck.el

```diff
@@ -3028,14 +3034,12 @@ Return ERRORS, modified in-place."
 
 Return t if ERR may be shown for the current buffer, or nil
 otherwise."
-  (flycheck-error-with-buffer err
-    (let ((file-name (flycheck-error-filename err))
-          (message (flycheck-error-message err)))
-      (and
-       (or (not file-name) (flycheck-same-files-p file-name (buffer-file-name)))
-       message
-       (not (string-empty-p message))
-       (flycheck-error-line err)))))
+  (let ((file-name (flycheck-error-filename err))
```

Comment:
I guess you can remove this binding, can't you?  As far as I can see this function doesn't access `filename` anymore, does it?